### PR TITLE
Add separate flood stage limit for frozen

### PR DESCRIPTION
**Pull request by henningandersen**
Dedicated frozen nodes can survive less headroom than other data nodes.
This commits introduces a separate flood stage threshold for frozen as
well as an accompanying max_headroom setting that caps the amount of
free space necessary on frozen.

Relates #71844

This PR builds on top of #71844, once that is merged in, those changes will be removed from this PR.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/unit/RelativeByteSizeValue.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    public boolean isAbsolute() {
+        return absolute != null;
+    }
+
+    public ByteSizeValue getAbsolute() {
+        return absolute;
+    }
+
+    public RatioValue getRatio() {
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/common/unit/RelativeByteSizeValue.java:43]: This entire class and associated test is also in #71844, only difference is that this method was made public here.

File: docs/reference/modules/cluster/disk_allocator.asciidoc [hunk 2]
```diff
@@ -113,6 +113,19 @@ PUT /my-index-000001/_settings
 --
 // end::cluster-routing-flood-stage-tag[]
 
+[[cluster-routing-flood-stage-frozen]]
+// tag::cluster-routing-flood-stage-tag[]
+`cluster.routing.allocation.disk.watermark.flood_stage.frozen` {ess-icon}::
+(<<dynamic-cluster-setting,Dynamic>>)
+Controls the flood stage watermark for frozen nodes, which defaults to 95%.
```
Comment (ywelsch) [inline docs/reference/modules/cluster/disk_allocator.asciidoc:120]: for "dedicated frozen nodes"?

**Review by ywelsch — COMMENTED** [review]
just 1 documentation comment, looking good o.w.

**Review by tlrx — APPROVED** [review]
LGTM

File: docs/reference/modules/cluster/disk_allocator.asciidoc (hunk 2 above)
Comment (henningandersen) [reply docs/reference/modules/cluster/disk_allocator.asciidoc]: Thanks, fixed in 22e49ed90dcbe78f18825328af8d7521ef0c06b3

**Comment by henningandersen** [comment]
@elasticmachine update branch

**Review by ywelsch — APPROVED** [review]
LGTM

**Pull request merged**